src/starknetCli.ts

```diff
@@ -97,39 +155,74 @@ async function compileCairoDependencies(
   const { success, resultPath } = compileCairo(filePath, path.resolve(__dirname, '..'), debug_info);
   if (!success) {
-    logError(`Compilation of contract ${filePath} failed`);
+    logCLIError(`Compilation of contract ${filePath} failed`);
     return;
   }
   console.log(`starknet-compile output written to ${resultPath}`);
 }
 
-export function runStarknetStatus(tx_hash: string, option: IOptionalNetwork) {
+export function checkStatus(tx_hash: string, option: IOptionalNetwork): string {
+  const options: Map<string, string | undefined> = new Map([
+    ['hash', tx_hash],
+    ['network', option?.network],
+  ]);
+
+  const cmd: string = buildCairoCommand(options, CAIRO_CMD_STATUS);
+
+  return cmd;
+}
+
+export function runStarknetStatus(tx_hash: string, option: IOptionalNetwork, isTest = false) {
+  if (tx_hash == undefined) {
+    logCLIError(`Error: Exception: tx_hash must be specified with the "tx_status" subcommand.`);
+    return;
+  }
   if (option.network == undefined) {
-    logError(
+    logCLIError(
       `Error: Exception: feeder_gateway_url must be specified with the "status" subcommand.\nConsider passing --network or setting the STARKNET_NETWORK environment variable.`,
     );
     return;
   }
 
-  try {
-    execSync(`${warpVenvPrefix} starknet tx_status --hash ${tx_hash} --network ${option.network}`, {
-      stdio: 'inherit',
-    });
-  } catch {
-    logError('starknet tx_status failed');
-  }
+  const cmd = checkStatus(tx_hash, option);
+  callCairoCommand(cmd, CAIRO_CMD_STATUS, true, isTest);
+
+  return cmd;
+}
+
+export function starkNetDeploy(
+  filePath: string,
+  option: IDeployProps,
+  inputs: string,
+  classHash: string | undefined,
+): string {
+  const wallet: string = option?.no_wallet
+    ? `--no_wallet --contract ${filePath}`
+    : option?.wallet === undefined
+    ? `${classHash}`
+    : `${classHash} --wallet ${option.wallet}`;
+
+  const options: Map<string, string | undefined> = new Map([
+    ['network', option?.network],
+    ['account', option?.account],
+  ]);
+
+  const multiOptions: string[] = [wallet, inputs];
+
+  const cmd: string = buildCairoCommand(options, CAIRO_CMD_STARKNET_DEPLOY, multiOptions);
+  return cmd;
 }
 
-export async function runStarknetDeploy(filePath: string, options: IDeployProps) {
+export async function runStarknetDeploy(filePath: string, options: IDeployProps, isTest = false) {
   if (options.network == undefined) {
-    logError(
+    logCLIError(
       `Error: Exception: feeder_gateway_url must be specified with the "deploy" subcommand.\nConsider passing --network or setting the STARKNET_NETWORK environment variable.`,
     );
     return;
   }
   // Shouldn't be fixed to warp_output (which is the default)
   // such option does not exists currently when deploying, should be added
-  const dependencyGraph = getDependencyGraph(filePath, 'warp_output');
+  const dependencyGraph: Map<string, string[]> = getDependencyGraph(filePath, 'warp_output');
```

Comment:
Can you also include the fix for the comment above? You'll need to pass the outputDir from the cli options

Reply:
option added and passed to functions